packages/amazon/src/serverGroup/configure/serverGroupCommandBuilder.service.js

```diff
@@ -213,15 +220,22 @@ angular
       function buildServerGroupCommandFromExisting(application, serverGroup, mode = 'clone') {
         const preferredZonesLoader = AccountService.getPreferredZonesByAccount('aws');
         const subnetsLoader = SubnetReader.listSubnets();
-
         const serverGroupName = NameUtils.parseServerGroupName(serverGroup.asg.autoScalingGroupName);
 
-        const instanceType = serverGroup.launchConfig
-          ? serverGroup.launchConfig.instanceType
-          : serverGroup.launchTemplate
-          ? serverGroup.launchTemplate.launchTemplateData.instanceType
-          : null;
-        const instanceTypeCategoryLoader = instanceTypeService.getCategoryForInstanceType('aws', instanceType);
+        let instanceTypes;
+        if (serverGroup.mixedInstancesPolicy) {
+          instanceTypes = serverGroup.mixedInstancesPolicy.launchTemplateOverridesForInstanceType
+            ? serverGroup.mixedInstancesPolicy.launchTemplateOverridesForInstanceType.map((o) => o.instanceType)
+            : serverGroup.mixedInstancesPolicy.launchTemplates[0].launchTemplateData.instanceType; // note: single launch template case is currently the only supported case for mixed instances policy
+        } else if (serverGroup.launchTemplate) {
+          instanceTypes = [serverGroup.launchTemplate.launchTemplateData.instanceType];
+        } else if (serverGroup.launchConfig) {
+          instanceTypes = [serverGroup.launchConfig.instanceType];
+        }
+        const instanceTypeCategoryLoader = instanceTypeService.getCategoryForMultipleInstanceTypes(
+          'aws',
+          instanceTypes,
+        );
```

Comment:
A general comment for this file. I see a lot of places with deeply nested values which is an opportunity for errors to be thrown. I would highly recommend lodash `get` for accessing these nested values since we know they are highly conditional.